src/mail-app/contacts/view/ContactGuiUtils.ts

```diff
@@ -201,3 +207,42 @@ export function compareContacts(contact1: Contact, contact2: Contact, sortByFirs
 		}
 	}
 }
+
+/**
+ * Check that we are allowed to sync contacts on an iOS device
+ * @param syncManager
+ * @returns false if no permission or iCloud sync is enabled and the user cancelled, or true if permission is granted and iCloud sync is disabled (or the user bypassed the warning dialog)
+ */
+export async function verifyContactSyncAllowedOnIos(syncManager: NativeContactsSyncManager): Promise<boolean> {
```

Comment:
This file is a very general contact GUI file and this is very specific to iOS and contact sync. Also because of that you now have loose locator import which is also bad. I think I would put it into sync manager or elsewhere.